java/src/main/java/com/google/appengine/tools/mapreduce/impl/util/RequestUtils.java

```diff
@@ -1,23 +1,59 @@
 package com.google.appengine.tools.mapreduce.impl.util;
 
+import com.google.appengine.tools.pipeline.impl.backend.AppEngineBackEnd;
+import com.google.appengine.tools.pipeline.impl.backend.AppEngineTaskQueue;
+import com.google.appengine.tools.pipeline.impl.backend.PipelineBackEnd;
 import com.google.cloud.datastore.Datastore;
 import com.google.cloud.datastore.DatastoreOptions;
 
 import jakarta.servlet.http.HttpServletRequest;
+import lombok.AllArgsConstructor;
 
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.Optional;
+
+/**
+ * handles translation to/from request parameters and pipeline backends
+ *
+ * q: good pattern? kinda franken factory to build instances from requests ... but w/o this, gets repeated in a bunch
+ *  of places
+ */
+@Singleton
+@AllArgsConstructor(onConstructor_ = @Inject)
 public class RequestUtils {
```

Comment:
This, this should be a web Filter, create the datastore instance in the filter and pass it through a request.setAttribute, then available everywhere.